File: algorithms/1_easy/C/03925_concatenate_array_with_reverse/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* concatWithReverse(const int* nums, const int len, int* return_len)
{
    assert(len > 0 && nums != NULL && len <= (INT_MAX / 2) && return_len != NULL);
    int* res = malloc((size_t)(2 * len) * sizeof(*res));
    assert(res != NULL);
    for (int i = 0; i < len; i += 1) {
        res[i] = nums[i];
        res[2 * len - 1 - i] = nums[i];
    }
    *return_len = 2 * len;
    return res;
}

File: algorithms/1_easy/C/03925_concatenate_array_with_reverse/solution.rs
impl Solution {
    pub fn concat_with_reverse(nums: Vec<i32>) -> Vec<i32> {
        let len = nums.len();
        assert!(len > 0);
        let mut res = Vec::with_capacity(2 * len);
        unsafe { res.set_len(2 * len); }
        for i in 0..len {
            res[i] = nums[i];
            res[2 * len - 1 - i] = nums[i];
        }
        res
    }
}
